src/pyk/kcfg/explore.py

```diff
@@ -178,6 +189,76 @@ def cterm_implies(
                 raise AssertionError(f'Received a non-substitution from implies endpoint: {subst_pred}')
         return CSubst(subst=Subst(_subst), constraints=ml_preds)
 
+    def implication_failure_reason(self, concrete: CTerm, abstract: CTerm) -> tuple[bool, str]:
+        def no_cell_rewrite_to_dots(term: KInner) -> KInner:
+            def _no_cell_rewrite_to_dots(_term: KInner) -> KInner:
+                if type(_term) is KApply and _term.is_cell:
+                    lhs = extract_lhs(_term)
+                    rhs = extract_rhs(_term)
+                    if lhs == rhs:
+                        return KApply(_term.label, [abstract_term_safely(lhs, base_name=_term.label.name)])
+                return _term
+
+            return bottom_up(_no_cell_rewrite_to_dots, term)
+
+        def _is_cell_subst(csubst: KInner) -> bool:
+            if type(csubst) is KApply and csubst.label.name == '_==K_':
+                csubst_arg = csubst.args[0]
+                if type(csubst_arg) is KVariable and csubst_arg.name.endswith('_CELL'):
+                    return True
+            return False
+
+        def _is_negative_cell_subst(constraint: KInner) -> bool:
+            constraint_bool = ml_pred_to_bool(constraint)
+            if type(constraint_bool) is KApply and constraint_bool.label.name == 'notBool_':
+                negative_constraint = constraint_bool.args[0]
+                if type(negative_constraint) is KApply and negative_constraint.label.name == '_andBool_':
+                    constraints = flatten_label('_andBool_', negative_constraint)
+                    cell_constraints = list(filter(_is_cell_subst, constraints))
+                    if len(cell_constraints) > 0:
+                        return True
+            return False
+
+        config_match = self.cterm_implies(CTerm.from_kast(abstract.config), CTerm.from_kast(concrete.config))
+        if config_match is None:
+            failing_cells = []
+            for cell in concrete.cells:
+                concrete_cell = concrete.cell(cell)
+                abstract_cell = abstract.cell(cell)
+                _LOGGER.info(f'concrete_cell: {concrete_cell}')
+                _LOGGER.info(f'abstract_cell: {abstract_cell}')
+                cell_match = abstract_cell.match(concrete_cell)
+                _LOGGER.info(f'cell_match: {cell_match}')
```

Comment:
I think we can remove these debugging statements.

```suggestion
                cell_match = abstract_cell.match(concrete_cell)
```